api/queue/queue.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2023 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package queue
+
+import (
+	"net/http"
+
+	"github.com/go-vela/server/router/middleware/user"
+	"github.com/go-vela/types/library"
+
+	"github.com/gin-gonic/gin"
+	"github.com/sirupsen/logrus"
+)
+
+// swagger:operation POST /api/v1/admin/queue-registration admin QueueRegistration
+//
+// Get queue credentials
+//
+// ---
+// produces:
+// - application/json
+// security:
+//   - ApiKeyAuth: []
+// responses:
+//   '200':
+//     description: Successfully generated registration token
+//     schema:
+//       "$ref": "#/definitions/Token"
+//   '401':
+//     description: Unauthorized
+//     schema:
+//       "$ref": "#/definitions/Error"
+
+// QueueRegistration represents the API handler to
+// provide queue credentials after worker onboarded with registration token.
+func QueueRegistration(c *gin.Context) {
+	// retrieve user from context
+	u := user.Retrieve(c)
+
+	logrus.Infof("Platform admin %s: fetching queue creds", u.GetName())
```

Comment:
Let's leverage the normal logging format that we use:
```go
logrus.WithFields(logrus.Fields{
		"user":  cl.Subject,
	}).Info("requesting queue credentials with registration token")
```

Note: you'll have to use claims rather than user